app.py

```diff
@@ -0,0 +1,470 @@
+    period = st.multiselect(
+        "開講年度・開講期",
+        [
+            "2024・前期",
+            "2024・後期",
+            "2024・通年",
+            "2024・前期集中",
+            "2024・後期集中",
+            "2024・通年集中",
+            "2024・前期不定",
+            "2024・後期不定",
+            "2024・通年不定",
+            "2024・前期前半",
+            "2024・前期後半",
+            "2024・後期前半",
+            "2024・後期後半",
+        ],
+    )
+    target = st.multiselect("対象学生", ["全学向", "文系向", "理系向", "留学生"])
+    esubject = st.multiselect("E科目", ["E1", "E2", "E3"])
+    keyword = st.text_input("キーワード")
+    professor = st.text_input("教授・教員")
+    grade = st.multiselect(
+        "成績評価方法", ["平常点", "課題", "発表", "討論", "小レポート", "小テスト", "期末レポート", "期末試験"]
+    )
+
+    graderatio = st.text_input("成績評価の占める最低パーセント", "0")
+
+    query = st.text_input("どんな講義をお探しですか？", "例: 機械学習について学べる科目は何ですか")
+
+    k = st.text_input("検索数", "4")
+
+    submit_btn = st.button("検索")
+    cancel_btn = st.button("キャンセル")
+
+    if submit_btn:
+        # 入力を統合する
+        store = FAISS.load_local("db/data2/syllabus_vectorstore", embedding, allow_dangerous_deserialization=True)
+        ids = []
+        if faculity != "---":
+            ids.append(search(faculity, "学部", metadatas))
+        if group != "---":
+            ids.append(search(group, "群", metadatas))
+        if len(department) != 0:
+            ids.append(or_list_search(department, departments))
+        if len(field) != 0:
+            ids.append(or_search(field, "分野", metadatas))
+        if len(yj) != 0:
+            ids.append(or_list_search(yj, yojigen))
+        if len(classtype) != 0:
+            ids.append(or_list_search(classtype, classtypes))
+        if len(language) != 0:
+            ids.append(or_search(language, "使用言語", metadatas))
+        if len(period) != 0:
+            ids.append(or_search(period, "開講年度・開講期", metadatas))
+        if len(target) != 0:
+            ids.append(or_search(target, "対象学生", metadatas))
+        if len(esubject) != 0:
+            ids.append(or_search(esubject, "E科目", metadatas))
+        if len(keyword) != 0:
+            ids.append(word_search(keyword, keywordtexts))
+        if len(professor) != 0:
+            ids.append(word_search(professor, professors))
+        if len(grade) != 0:
+            ids.append(hyoka_search(grade, int(graderatio), hyoka))
+        if len(ids) != 0:
+            store.delete(ids_union(ids))
+        
+
+        # 質問で検索するための検索候補の個数を求める
+        ID = [str(i) for i in range(len(texts))]
+        if len(ids) != 0:
+            for i in ids_union(ids):
+                ID.remove(i)
+        st.text("検索候補は" + str(len(ID)) + "個")
+
+        if len(ID) > 0:
+            # 類似度が高い順に科目を検索する
+            a = store.similarity_search(query, k=int(k))
+            st.text("該当する科目は以下の通りです。")
+            # 検索した科目を列挙する
+            for i in range(len(a)):
+                st.text(a[i].metadata["科目名"] + "   URL:" + a[i].metadata["URL"])

```

Comment:
link_buttonを使ってください！
https://docs.streamlit.io/develop/api-reference/widgets/st.link_button

```suggestion
                st.link_button(label=a[i].metadata["科目名"], url= a[i].metadata["URL"])
```